classes/form/cms_types_form.php

```diff
@@ -90,6 +92,20 @@ protected function definition() {
         $this->add_action_buttons();
     }
 
+    /**
+     * Put together the datasource selection mechanism using checkboxes.
+     */
+    public function add_datasource_select_element() {
+        $mform = $this->_form;
+        $labels = dsbase::get_datasource_labels();
+        $mform->addElement('static', 'datasources_desc', get_string('datasources', 'mod_cms'),
```

Comment:
I think the better way to do this is inside a group so it is a single element, example here:

https://github.com/danmarsden/moodle-mod_attendance/blob/MOODLE_402_STABLE/classes/form/addsession.php#L161-L174

There is a few examples in core too

Reply:
This done, although it puts them all into a single line.